src/web/topic/store/utilActions.ts

```diff
@@ -37,6 +37,16 @@ export const setTopicData = (state: TopicStoreState, sessionUsername?: string) =
   emitter.emit("loadedTopicData", getTopicDiagram(useTopicStore.getState()));
 };
 
+export const getSelectedEdge = () => {
```

Comment:
@keyserj is this sort of what you had in mind? as an aside what do you think about having some common interface for both nodes/edges in order to deduplicate some logic?

Reply:
This is the right logic, but I think a hook is more suitable than an action here because an action doesn't "watch" for changes to re-render, so the `disabled` check can become outdated when selection changes.

It seems to be working right now because there's another hook triggering a re-render when selection changes - I think it's [useTemporalHooks](https://github.com/amelioro/ameliorate/blob/c1458cc15e51e1e1bd15fc079df308f2d1f30fb7/src/web/topic/components/Surface/TopicToolbar.tsx#L63) since undo/redo currently include changes to selection (which is not ideal but is how it works right now ⚰️). It's fine now but if we were to extract undo/redo into a separate component, selection changes would no longer re-render the whole toolbar along with the `disabled` check.

(I also thought that hook logic would only run when the store changes, saving some performance over an action, but it appears that hook selectors run on all re-renders... I'll look into this more separately)

Secondly, I think using a common interface for nodes/edges is probably worthwhile, since I think there will be other actions that will apply to both nodes & edges.

So my thought is that we could make something like `useSelectedGraphPart` (or "parts" to be more future-proof when we can select multiple, but that might be more work). Probably would live in `graphPartHooks.ts`. Then when we use the graph part in the button's `onClick`, we could invoke a new `deleteGraphPart(graphPart)` action that checks if the graph part is a node/edge (probably easiest to just check [type === "ScoreEdge"](https://github.com/amelioro/ameliorate/blob/c1458cc15e51e1e1bd15fc079df308f2d1f30fb7/src/web/topic/utils/diagram.ts#L79) for now) and calls the specific delete method.

Reply:
cool, TIL about hooks. I've made the changes and I've made a `GraphPart` interface for now. could consider having Node and Edge extend `GraphPart` (and maybe making its type be `FlowNodeType | <whatever types for edges>)` too, but maybe not necessary for now